### ISSUE TITLE
Use OpenPifPaf for videos

### ISSUE THREAD

**Issue by Christoph9402** [openpifpaf/openpifpaf#417]
Hello, I am working on a project, where I want to implement OpenPifPaf. The general structure of my code is as follows:
```
import libaries

net_cpu, _ = openpifpaf.network.Factory(checkpoint='shufflenetv2k30-201104-224654-cocokp-d75ed641.pkl', download_progress=True).factory()
net = net_cpu.to(device)
decoder = openpifpaf.decoder.factory([hn.meta for hn in net_cpu.head_nets])
openpifpaf.decoder.CifCaf.force_complete = True
preprocess =openpifpaf.transforms.Compose([openpifpaf.transforms.NormalizeAnnotations(),openpifpaf.transforms.CenterPadTight(16),openpifpaf.transforms.EVAL_TRANSFORM])
annotation_painter = openpifpaf.show.AnnotationPainter(xy_scale=1.0)
openpifpaf.show.Canvas.show = True

#load video
cap = cv2.VideoCapture(file)
while(True):
    #extract every frame from video
    ret, frame = cap.read()

###############
   #other operations
###############

   #find pose in video
    data = openpifpaf.datasets.PilImageList([image], preprocess=preprocess)
    loader = torch.utils.data.DataLoader(data, batch_size=1, pin_memory=True, collate_fn=openpifpaf.datasets.collate_images_anns_meta)
    
    for images_batch, _, __ in loader:
        predictions = decoder.batch(net, images_batch, device=device)[0]
        with openpifpaf.show.image_canvas(cropped_image) as ax:
            annotation_painter.annotations(ax, predictions)
```
I am able to extract the pose of persons inside the frame, but it is very slow that way. I already checked, that Cuda was enabled. When I run the webcam script using `python -m openpifpaf.video --source 0 --show --video-fps 30 --instance-threshold 0.0` I am able to get around 12 FPS, which would be great for my project. With the code structure I am using currently I get less than 1FPS. Is there an easy way of applying OpenPifPaf to every frame, where I can get the same framerate as with the webcam demo? Since there are also other algorithms apllied, I must use OpenPifPaf for every frame and am not able to load in just a single mp4 file.
Also, is there a way to save the Output to a new video, or do I have to save every frame as a .png?
Thank you :)

**Comment by Christoph9402** [comment]
Hello @svenkreiss, Thank you for your answer. I did a few tests. When I don't use Matplotlib to show the results, the time needed to find the pose on the image with original size was around 0.33s. The input size of the image is 1920*540px. After using RescaleAbsolute(1281,fast=True), the time needed was reduced to 0.15s. Since the images I want to apply OpenPifPaf on are from the PIE Dataset (camera from a moving vehicle), I want to make sure that Persons in the distance are detected while maintaining the highest speed possible. Resizing was already a great tipp. :)
Could the inferencing time be reduced even further with a different model? If so, could you recomment one for my use case? Looking at the table, where pretrained models are compared, it seems like a shufflenet2k16 could be even faster, right? Are there any results for the models that are [released](https://github.com/vita-epfl/openpifpaf-torchhub/releases)?I think that I used the model which was used for Monoloco.

**Comment by svenkreiss** [comment]
There is this table https://openpifpaf.github.io/intro.html#pre-trained-models but I think you already found that.
There are also example images with predicted poses: https://openpifpaf.github.io/examples.html
What other results do you like to see?

**Comment by Christoph9402** [comment]
Thank you for your help @svenkreiss  :) Yes, I already found the table. I was just wondering if the performance metrics in that table are always the ones for the newest released model. I'm going to try out a few different model and imagesize combinations to find the one that is best for me. I'm going to close this topic for now. I could already see a good improvement in speed but I'll have to try a bit more to find the right combination of speed and accuracy.
